core/src/main/scala/quasar/fs/Natural.scala

```diff
@@ -48,4 +56,8 @@ object Natural {
     Monoid.instance(
       (x, y) => Mult(Mult.unwrap(x) * Mult.unwrap(y)),
       Mult(_1))
+
+  implicit val equal: Equal[Natural] = Equal.equalBy(_.value)
```

Comment:
Since we do define the `equals` method, we should use `Equal.equalA` here to ensure consistency.